tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -854,16 +855,121 @@ impl<'a, F> fmt::Debug for FieldFnVisitor<'a, F> {
     }
 }
 
+// === printing synthetic Span events ===
+
+/// Select the degree to which tracing spans are logged as events
+#[derive(Debug)]
+pub enum FmtSpan {
+    /// spans are ignored (this is the default)
+    None,
+    /// one event per enter/exit of a span
+    Active,
+    /// one event when the span is dropped
+    Close,
+    /// events at all points (new, enter, exit, drop)
+    Full,
+}
+
+pub(super) struct FmtSpanConfig {
+    pub(super) kind: FmtSpan,
+    pub(super) fmt_timing: bool,
+}
+
+impl FmtSpanConfig {
+    pub(super) fn without_time(self) -> Self {
+        Self {
+            kind: self.kind,
+            fmt_timing: false,
+        }
+    }
+    pub(super) fn with_kind(self, kind: FmtSpan) -> Self {
+        Self {
+            kind,
+            fmt_timing: self.fmt_timing,
+        }
+    }
+    pub(super) fn fmt_timing(&self, t: u64, buf: &mut impl io::Write) {
+        if self.fmt_timing {
+            fmt_nanos(t, buf)
+        }
+    }
+    pub(super) fn trace_new(&self) -> bool {
+        match self.kind {
+            FmtSpan::None => false,
+            FmtSpan::Active => false,
+            FmtSpan::Close => false,
+            FmtSpan::Full => true,
+        }
+    }
+    pub(super) fn trace_enter(&self) -> bool {
+        match self.kind {
+            FmtSpan::None => false,
+            FmtSpan::Active => true,
+            FmtSpan::Close => false,
+            FmtSpan::Full => true,
+        }
+    }
+    pub(super) fn trace_exit(&self) -> bool {
+        match self.kind {
+            FmtSpan::None => false,
+            FmtSpan::Active => true,
+            FmtSpan::Close => false,
+            FmtSpan::Full => true,
+        }
+    }
+    pub(super) fn trace_close(&self) -> bool {
+        match self.kind {
+            FmtSpan::None => false,
+            FmtSpan::Active => false,
+            FmtSpan::Close => true,
+            FmtSpan::Full => true,
+        }
+    }
+}
+
+impl Debug for FmtSpanConfig {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.kind.fmt(f)
+    }
+}
+
+impl Default for FmtSpanConfig {
+    fn default() -> Self {
+        Self {
+            kind: FmtSpan::None,
+            fmt_timing: true,
+        }
+    }
+}
+
+fn fmt_nanos(t: u64, buf: &mut impl io::Write) {
+    let mut t = t as f64;
+    for unit in ["ns", "µs", "ms", "s"].iter() {
+        if t < 10.0 {
+            let _ = write!(buf, "{:.2}{}", t, unit);
+            return;
+        } else if t < 100.0 {
+            let _ = write!(buf, "{:.1}{}", t, unit);
+            return;
+        } else if t < 1000.0 {
+            let _ = write!(buf, "{:.0}{}", t, unit);
+            return;
+        }
+        t /= 1000.0;
+    }
+    let _ = write!(buf, "{:.0}s", t * 1000.0);
+}
```

Comment:
@hawkw I couldn’t find such functionality in chrono; let me know if you know of a better alternative.

Reply:
this is fine, then!